docs/applications/go.md

```diff
@@ -0,0 +1,42 @@
+# Go
+
+## Compiling your application
+
+Since Go 1.21, you can natively compiled Go applications to WebAssembly using the wasip1 GOOS:
+
+```bash
+GOOS=wasip1 GOARCH=wasm go build -o app.wasm <path/to/main/pkg>
+```
+
+If you prefer using TinyGo:
+
+```bash
+tinygo build -o app.wasm -target=wasi <path/to/main/pkg>
+```
+
+This will build a WebAssembly module that can be run with Timecraft.
+
+## Running your application with Timecraft
+
+To run your application:
+
+```bash
+timecraft run app.wasm
+```
+
+Command-line arguments can be specified after the WebAssembly module. To prevent
+Timecraft from interpreting command-line options for the application, use:
+
+```bash
+timecraft run -- app.wasm arg1 arg2 ...
+```
```

Comment:
How about adding a note that `--` is optional but a good practice to explicitly separate the arguments to `timecraft run` from those passed to the module?